experimental/dds/tree-graphql/src/graphql-plugins/SharedTreePlugin.ts

```diff
@@ -0,0 +1,330 @@
+		return printResolver(fieldName, getNodeID);
+	}
+
+	// Check if this field is a grahpql scalar type, and decode appropriately
+	if (isScalarTypeName(fieldInfo.typeName)) {
+		return printScalarResolver(fieldName, fieldInfo);
+	}
+
+	// Check if this field is an enum type. If so, decode as a string
+	if (schema.getType(fieldInfo.typeName)?.astNode?.kind === 'EnumTypeDefinition') {
+		return printScalarResolver(fieldName, { ...fieldInfo, typeName: 'String' });
+	}
+
+	// This field is a non-leaf node, so just return the id of the node to be used by sub nodes
+	switch (fieldDef.type.kind) {
+		case 'NamedType':
+			return printResolver(fieldName, getTrait);
+		case 'NonNullType':
+			return printResolver(fieldName, getNonNullTrait);
+		case 'ListType':
+			return printResolver(fieldName, getListTrait);
+		default:
+			throw Error(`Unrecognized or unsupported field type: ${fieldDef.type}`);
+	}
+}
+
+function printScalarResolver(fieldName: string, scalar: FieldInfo): string {
+	if (scalar.isList === true) {
+		switch (scalar.typeName) {
+			case 'String':
+				return printResolver(fieldName, getStringList);
+			case 'Float':
+				return printResolver(fieldName, getFloatList);
+			case 'Int':
+				return printResolver(fieldName, getIntList);
+			case 'Boolean':
+				return printResolver(fieldName, getBooleanList);
+			case 'ID':
+				return printResolver(fieldName, getIDList);
+			default:
+				throw Error(`Unrecognized scalar type: ${scalar.typeName}`);
+		}
+	}
+
+	switch (scalar.typeName) {
+		case 'String':
+			return printResolver(fieldName, getString);
+		case 'Float':
+			return printResolver(fieldName, getFloat);
+		case 'Int':
+			return printResolver(fieldName, getInt);
+		case 'Boolean':
+			return printResolver(fieldName, getBoolean);
+		case 'ID':
+			return printResolver(fieldName, getID);
+		default:
+			break;
+	}
+
+	throw Error(`Unrecognized scalar type: ${scalar.typeName}`);
+}
+
+function printResolver<TSource, TArgs>(fieldName: string, resolver: GraphQLFieldResolver<TSource, TArgs>): string {
+	return `${fieldName}: ${resolver.name},`;
+}
+
+/** Digs into a field node in the AST and determines if it a list and/or non-null */
+function unwrapField(fieldType: TypeNode, isNonNull?: boolean, isList?: boolean): FieldInfo {
+	switch (fieldType.kind) {
+		case 'NonNullType':
+			return unwrapField(fieldType.type, true, isList);
+		case 'ListType':
+			return unwrapField(fieldType.type, isNonNull, true);
+		case 'NamedType':
+			return {
+				typeName: fieldType.name.value,
+				isNonNull,
+				isList,
+			};
+		default:
+			throw Error(`Unrecognized or unsupported field type: ${fieldType}`);
+	}
+}
+
+function isScalarTypeName(typeName: string): boolean {
+	switch (typeName) {
+		case 'String':
+		case 'Float':
+		case 'Int':
+		case 'Boolean':
+		case 'ID':
+			return true;
+
+		default:
+			break;
+	}
+
+	return false;
+}
+
+// ##############################
```

Comment:
Eventually, we should split this into two packages:
* 'tree-graphql' which has runtime code referenced by the codegen'ed output
* 'graphql-codegen-fluid' which has the plugin
 
(The codegen tool scans the installed packages for plugins by matching the 'graphql-codegen-*' prefix)